nixos/modules/services/x11/unclutter.nix

```diff
@@ -0,0 +1,33 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let cfg = config.services.unclutter;
+in {
+  options = {
+    services.unclutter.enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Enable unclutter to hide your mouse cursor when inactive";
+    };
+
+    services.unclutter.arguments = mkOption {
+      description = "Arguments to pass to unclutter command";
+      default = "-idle 1";
+      type = types.uniq types.string;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.unclutter = {
+      description = "unclutter";
+      requires = [ "display-manager.service" ];
+      after = [ "display-manager.service" ];
+      wantedBy = [ "graphical.target" ];
```

Comment:
Not that this service won't start if you rely on graphical.target and xserver is not in autorun (so e.g. doing startx manually somehow).